sherpa/astro/xspec/tests/test_xspec.py

```diff
@@ -805,3 +816,37 @@ def test_apec_redshift_parameter_is_case_insensitive():
     mdl.REDshift = 0.2
     assert mdl.redshift.val == 0.2
     assert mdl.Redshift.val == 0.2
+
+
+@requires_xspec
+def get_xspec_models_with_exp_cache():
+    """Return the XSPEC models that use the modelCacher1d_exp cache.
+
+    Currently, that is: Models that inherit from XSAbsorptionModel.
+    """
+    models = get_xspec_models()
+    return [m for m in models if issubclass(m, XSAbsorptionModel)]
+
+
+@requires_xspec
+@pytest.mark.parametrize("modelcls", get_xspec_models_with_exp_cache())
+def test_models_with_exp_cache(modelcls):
+    '''Tests that models with the modelCacher1d_exp work correctly.
+
+    Unlike modelCacher1d, does not evaluate the model for every
+    parameter value once, but modelCacher1d_exp assumes a certain
+    functional form and calcualtes model values by itself, this this
+    warrantes a test for every model using this cache.
+    '''
+    x = np.arange(0.1, 10, 0.1)
```

Comment:
I was going to suggest decreasing the upper bound since we know not much happens >~ 2 keV, but I just tried with the tbabs model and there's some interesting going on around 7 keV we might want to ensure we catch.

We could decrease the bin width though - here's a comparison of the tbabs model with de=0.1 (top) and de=0.01 (bottom) - it might be nice to check we catch the wiggles well (or get a better handle on how well we can catch them)?

![sherpa](https://github.com/user-attachments/assets/c739f888-8873-44ab-8cbf-fc02275e1a71)

Reply:
The "wiggles" you talk about here arer the physical absorption edges. If you look closely, you'll the edges fo C, N, O, Si, Fe and and probably a few others at different energies. Those are not affected by this method. We don't interplotate over wavelength, but just over N_H. Just as in all other applications of our caching code, the `xlo,xhi` are part of the key of the cache. If you call with a different set of `xlo,xhi` the orininal model will be evaluated (and cached) again.

The only thing we need to check is that the orange curve is not identical to 0 or 1 in the plot above.